src/charm.py

```diff
@@ -64,6 +66,92 @@ def evaluate(self, csr: bytes, relation_id: int, requirer_csrs: list[RequirerCSR
         return True
 
 
+class AllowedFields:
+    """Filter the CSR so as to only allow CSRs that match the given regexes for the CSR fields."""
+
+    EMAIL_SUBJECT_NAME = "1.2.840.113549.1.9.1"
+    ORGANIZATION_SUBJECT_NAME = "O"
+    COUNTRY_CODE_SUBJECT_NAME = "C"
+
+    def __init__(self, filters: dict):
+        self.field_filters = filters
+
+    def evaluate(self, csr: bytes, relation_id: int, requirer_csrs: list[RequirerCSR]) -> bool:  # noqa: C901
```

Comment:
To get rid of the C901 code, we could have smaller methods for each check and just call them if the filter is set, but this is really minor.

Reply:
I did try a couple of ways to split them up, but they just ended up being less readable in my opinion.

Reply:
Ok, let's go with this for now.